cmd/config-reloader/main.go

```diff
@@ -60,6 +65,39 @@ func main() {
 		level.Error(logger).Log("msg", "parsing reloader URL failed", "err", err)
 		os.Exit(1)
 	}
+
+	// Set up interrupt signal handler.
+	term := make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
+
+	// Poll ready endpoint indefinitely until it's up and running.
+	req, _ := http.NewRequest(http.MethodGet, *readyURLStr, nil)
```

Comment:
Let's capture the error of `http.NewRequest` and exit early if it happens.

Reply:
Done.